worker-batch-ingestion-plugin/src/test/java/com/github/cafdataprocessing/worker/ingestion/DocumentWorkerDocumentDeserializerTest.java

```diff
@@ -0,0 +1,139 @@
+        documentWorkerDocument.failures.add(getDocumentWorkerFailure());
+
+        documentWorkerDocument.subdocuments = new ArrayList<>();
+        for(int i = 0; i <1000; i ++){
+            documentWorkerDocument.reference = String.format("r%s", i);
+            final DocumentWorkerDocument subdocument = new DocumentWorkerDocument();
+            subdocument.reference = "subdocument " + i;
+            subdocument.fields = new HashMap<>();
+            subdocument.fields.put("myfield", getDocumentWorkerFieldValues());
+
+            subdocument.failures = new ArrayList<>();
+            subdocument.failures.add(getDocumentWorkerFailure());
+            
+            documentWorkerDocument.subdocuments.add(subdocument);
+        }
+
+        final DocumentWorkerDocumentTask documentWorkerDocumentTask = new DocumentWorkerDocumentTask();
+        documentWorkerDocumentTask.document = documentWorkerDocument;
+
+        final String json = objectMapper.writeValueAsString(documentWorkerDocumentTask);
+        
+        final DocumentWorkerDocumentTask deserialisedDocumentWorkerDocumentTask = 
+                objectMapper.readValue(json, DocumentWorkerDocumentTask.class);
+        
+        assertEquals(100, deserialisedDocumentWorkerDocumentTask.document.subdocuments.size());
+        assertEquals(2, deserialisedDocumentWorkerDocumentTask.document.failures.size());
+        
+        final List<DocumentWorkerFailure> actualFailures = 
+                deserialisedDocumentWorkerDocumentTask.document.failures.stream()
+                .filter(f -> f.failureId.equals("IBWP-SUBDOCUMENTS_TRUNCATED-WARNING")).collect(Collectors.toList());
+        
+        assertEquals(1, actualFailures.size());
+        final DocumentWorkerFailure truncationFailure = actualFailures.get(0);
+        assertEquals("Subdocuments were truncated at 100", truncationFailure.failureMessage);
+        assertNotNull(truncationFailure.failureStack);
+        
+    }
+
+    @Test
+    void deserializeInvalidDocumentTest() throws JsonProcessingException {
+
+        final String json = "[]";
+        try {
+            final DocumentWorkerDocument documentWorkerDocument =
+                    objectMapper.readValue(json, DocumentWorkerDocument.class);
+        }
+        catch(final IllegalStateException ex) {
+            assertEquals("Expected '{' at [Source: (String)\"[]\"; line: 1, column: 2]", ex.getMessage());
+        }
+        
+    }
+
+    @Test
+    void deserializeInvalidSubdocumentsTest() throws JsonProcessingException {
+
+        final String json = "{\"subdocuments\":{}}";
+
+        try {
+            final DocumentWorkerDocument documentWorkerDocument =
+                    objectMapper.readValue(json, DocumentWorkerDocument.class);
+        }
+        catch(final IllegalStateException ex) {
+            assertEquals("Expected '[' at [Source: (String)\"{\"subdocuments\":{}}\"; line: 1, column: 18]", ex.getMessage());
+        }
+
```

Comment:
Looks good, I've no issues with it other than it passes if the Json Exception is thrown and not tested for in the test and local variable not used. I'd write it this way to only allow test to pass if IllegalState thrown
    @Test
    void deserializeInvalidSubdocumentsTest() {

        final String json = "{\"subdocuments\":{}}";
        IllegalStateException thrown = Assertions.assertThrows(IllegalStateException.class, () -> {
            objectMapper.readValue(json, DocumentWorkerDocument.class);
        });
        Assertions.assertEquals("Expected '[' at [Source: (String)\"{\"subdocuments\":{}}\"; line: 1, column: 18]",
                thrown.getMessage());
    }